### PR TITLE
CI Standardisation

### PR DESCRIPTION
**Pull request by jmsmkn**
This PR:
- Migrates to PEP517
- Standardises README and HISTORY formats to markdown
- Removes codecov (didn't work, we can discuss how to handle that service)

### CODE REVIEW

**Comment by chrisvanrun** [comment]
Seem like the tox env is mixing up the python version calls (i.e. py38 is being called in py37)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,48 +1,73 @@
...
-    needs: [precommit]
+  build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         python: ["3.6", "3.7", "3.8", "3.9", "3.10"]
     steps:
       - uses: actions/checkout@v2
-      - name: Setup Python
+      - name: Set up Python ${{ matrix.python-version }}
```
Comment (chrisvanrun) [inline .github/workflows/ci.yml:41]: Possibly this should be `matrix.python`?

Comment (jmsmkn) [reply .github/workflows/ci.yml:41]: Good catch! The usage seems OK, but this also happens on other repos where we test for 3.10 and use strings for the python version.

Comment (jmsmkn) [reply .github/workflows/ci.yml:41]: You were right! Should be resolved now.

**Review by chrisvanrun — APPROVED** [review]
LGTM!

**Pull request merged**